### ISSUE TITLE
new event to indicate that the column setup (getColumnSetup) has changed

### ISSUE THREAD

**Issue by StephanWald** [BBj-Plugins/BBjGridExWidget#72]
The UI developer is interested in knowing if the user has changed column order or sizing, so he can enable a "save" button for the user to store the current setting (or warn before leaving the program if he wants to save or abandon the new grid setup)

**Comment by hyyan** [comment]
@StephanWald 

The grid supports a long list of events [here](https://www.ag-grid.com/javascript-grid-events/#columns) and [here](https://www.ag-grid.com/javascript-grid-events/#sort-filter) . In theory, we can support them all. but we need to test carefully the performance of these events in BBj, some of these events might be fired 10+ time in one millisecond unless we denounce them.

I think the UI developer can still achieve this kind of monitoring by using `getColumnState` method, the method returns a string, which means you can get the initial state and compare later.

Does this seem as a solution for you ?

**Comment by StephanWald** [comment]
The purpose of this request is to enable a "save" button or to ask the user if (s)he wants to save the current status whenever getColumnState has changed versus the initial load. So the request is basically about watching the output of getColumnState.

**Comment by StephanWald** [comment]
please review again, we need a solution for this. We only need one event that is fired when the column order, visibility or sizing is changed, or the filter settings are changed.
Can the Javascript side maybe do the following:
- catch all these events
- delay the reporting to BBj by one second
- if another event comes in from ag-grid before this delay is over, restart the delay time to be again one second from the last event
That way there is a maximum of one event per second (maybe we make it two?) reported to the BBj side, so BBj can enable a "save settings" button or auto-save the state, and we avoid hammering the event flow.

**Comment by StephanWald** [comment]
introduced ON_GRID_COLUMN_STATE_CHANGE event, when this is registered a timer checks the grid setup every second. Ugly workaround until we can trigger the event from javascript

**Comment by hyyan** [comment]
@StephanWald This issues is fixed now in the dev branch.
I have reverted your last commit 7b8c97b6142b4a7436f84266dfc52960f18ecf90 (only dev) but kept the new event name `ON_GRID_COLUMN_STATE_CHANGE` .

`ColumnStateDemo.bbj` has an example showing how to monitor the state correctly. 
The state is considered dirty when ag grid fires one of the following events:
```
'sortChanged',
'filterChanged',
'columnVisible',
'columnPinned',
'columnResized',
'columnMoved',
'newColumnsLoaded',
'gridColumnsChanged',
'displayedColumnsChanged',
'virtualColumnsChanged',
'columnEverythingChanged',
'gridSizeChanged',
'expandOrCollapseAll',
'toolPanelVisibleChanged'
```
BBj will be notifed once per change , all these events are debounce in Javascript to avoid overloading BBj with the insane change state calls.

Closing this one as I consider it resolved, please re-open in case there are any issues left regarding this one.

The master branch will receive this update when the dev branch is merged back into the master

**Comment by hyyan** [comment]
@StephanWald  The fix is available now in v0.92.0
